youtube_dl/extractor/sverigesradio.py

```diff
@@ -0,0 +1,109 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    int_or_none,
+    try_get,
+)
+
+
+class SverigesRadioBaseIE(InfoExtractor):
+    _BASE_URL = 'https://sverigesradio.se/sida/playerajax'
+
+    def _extract_audio(self, audio_type, secondary_type, audio_id):
+        query = {'id': audio_id, 'type': audio_type}
+
+        metadata = self._download_json(self._BASE_URL + '/audiometadata',
+            audio_id, query=query)
+        item = metadata['items'][0]
+
+        formats = []
+
+        audiourl = self._download_json(
+            self._BASE_URL + '/getaudiourl',
+            audio_id, query=dict(query, type=audio_type, quality='high', format='iis'))
+        formats.append({'url': compat_str(audiourl['audioUrl']), 'vcodec': 'none'})
+
+        if secondary_type is not None:
+            secondaryurl = self._download_json(
+                self._BASE_URL + '/getaudiourl',
+                audio_id, fatal=False,
+                query=dict(query, type=secondary_type, quality='high', format='iis'))
+            if secondaryurl['audioUrl']:
+                formats.append({'url': compat_str(secondaryurl['audioUrl']), 'vcodec': 'none'})
```

Comment:
I guess this could be generalized a bit into list of types with name and fatal=true/false. But worth it?

Reply:
I've no idea what you're about. But this looks fine apart from proper `url` extraction.

Reply:
Ok! better now?

Reply:
Seems like the API repsonse has changed to empty string url also